src/routes/NewBoard/NewBoard.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { useFirestore } from 'react-redux-firebase';
+import Input from '@material-ui/core/Input';
+import Button from '@material-ui/core/Button';
+import {AuthenticationManager} from 'utils/authentication/AuthenticationManager';
+import BoardData from "types/BoardData";
+import firebase from 'firebaseSetup';
+
+import getRandomName from 'constants/Name';
+import {RouteComponentProps} from "react-router";
+
+interface NewBoardProps extends RouteComponentProps {}
+
+function NewBoard(props: NewBoardProps) {
+
+    const firestore = useFirestore();
+
+    let auth = firebase.auth().currentUser;
+
+    const [name, setName] = React.useState(getRandomName());
+    function handleChangeName(e: any) {
+        const name = (e.target as HTMLInputElement).value;
+        setName(name);
+    }
+
+    function handleLogin() {
+        AuthenticationManager.signInAnonymously(name).then(_ => {
```

Comment:
`signInAnonymously` should return a `Task` of the `AuthResult`, and therefore you should be able to write ((task) => { task.getResult().getUser() }) without a second call on `firebase`. Dunno how this is handled internally, but the way with the param seems "safer" to me